### PR TITLE
Fix transaction size comments. Size now refers to virtual size (BIP141).

### PR DESCRIPTION
**Pull request by jnewbery**
As part of BIP 141, transaction sizes have been replaced by virtual size (with witness data discounted). Code comments and rpc help text have not been updated everywhere to reflect this change.

This PR fixes various comments and rpc help text.

### CODE REVIEW

**Comment by jnewbery** [comment]
I've built and tested the changes.

File: src/rpc/mining.cpp [hunk 1]
```diff
@@ -779,8 +779,8 @@ UniValue estimatefee(const UniValue& params, bool fHelp)
     if (fHelp || params.size() != 1)
         throw runtime_error(
             "estimatefee nblocks\n"
-            "\nEstimates the approximate fee per kilobyte needed for a transaction to begin\n"
-            "confirmation within nblocks blocks.\n"
+            "\nEstimates the approximate fee per virtual kilobyte needed for a transaction to begin\n"
```
Comment (sdaftuar) [inline src/rpc/mining.cpp]: nit: I might drop the word virtual here?  I think the last sentence explains sufficiently...

**Review by sdaftuar — COMMENTED** [review]
One nit, but otherwise looks good to me.

As a matter of good practice, in addition to building your changes locally, you should also locally run the tests (make check + qa/pull-tester/rpc-tests.py) to make sure nothing inadvertently broke.

File: src/rpc/blockchain.cpp [hunk 2]
```diff
@@ -331,18 +331,18 @@ UniValue getdifficulty(const UniValue& params, bool fHelp)
 
 std::string EntryDescriptionString()
 {
-    return "    \"size\" : n,             (numeric) transaction size in bytes\n"
+    return "    \"size\" : n,             (numeric) virtual transaction size. Differs from size for witness transactions since witness size is discounted. BIP 141.\n"
```
Comment (TheBlueMatt) [inline src/rpc/blockchain.cpp]: This doesnt elucidate what this is to me -  maybe the second sentence should be like "Differs from size for witness transactions as this does not include the witness data" or something like that (I dont actually know what its supposed to be).

Comment (sdaftuar) [reply src/rpc/blockchain.cpp]: His language is very similar to the explanatory text for "vsize" used in the rawtransaction rpc's.  Unfortunately, BIP 141 doesn't define virtual transaction size, but I think the best way to wrap this up would be to use the term virtual size here, and then suggest that they update the BIP to define it as well.

Here, virtual size refers to 1/4 of the transaction weight (so base_tx_bytes + witness_size/4).

Comment (jonnynewbs) [inline src/rpc/blockchain.cpp]: How about "virtual transaction size. This is different from actual serialized size for witness transactions as witness data is discounted (BIP 141)"

File: src/rpc/mining.cpp (hunk 1 above)
Comment (jnewbery) [reply src/rpc/mining.cpp]: ACK

File: src/rpc/blockchain.cpp (hunk 2 above)
Comment (jnewbery) [reply src/rpc/blockchain.cpp]: @sdaftuar I've opened a PR against the BIPS repository: https://github.com/HackerResidency/bips/pull/1

Comment (TheBlueMatt) [reply src/rpc/blockchain.cpp]: Seems reasonable, though you might have write "See BIP 141" instead of just "BIP 141" :).

Comment (jnewbery) [reply src/rpc/blockchain.cpp]: Sure thing. Please see 73b772d

**Review by sdaftuar — APPROVED** [review]
Looks good to me.  Just make sure you squash these down to one commit before opening the PR upstream.

Since you reference definitions in the BIP, you may want to open the BIP PR first (I'll review that now) and then in the PR text for this pull, link to your BIP 141 update PR as well.

**Review by TheBlueMatt — APPROVED** [review]

**Comment by jnewbery** [comment]
I've opened a pull request to bitcoin/bitcoin. Closing this PR.

**Pull request closed without merging**